include/signer/signer.hpp

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <future>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "data/data.hpp"
+
+namespace nostr
+{
+namespace signer
+{
+enum class Encryption
+{
+    NIP04,
+    NIP44
+};
+
+/**
+ * @brief An interface for Nostr event signing that implements NIP-46.
+ */
+class ISigner
+{
+public:
+    virtual ~ISigner() = default;
+
+    /**
+     * @brief Signs the given Nostr event.
+     * @param event The event to sign.
+     * @returns A promise that will be fulfilled when the event has been signed.  It will be
+     * fulfilled with `true` if the signing succeeded, and `false` if it failed.
+     * @remark The event's `sig` field will be updated in-place with the signature.
+     */
+    virtual std::shared_ptr<std::promise<bool>> sign(
+        std::shared_ptr<nostr::data::Event> event
+    ) = 0;
+};
+
+class INostrConnectSigner : public ISigner
```

Comment:
Should we add some method to store and check the status of a connection?

Reply:
That would be a ping method, probably.  That would make sense.  Give consumers a way to check whether the connection is active.

When you say store, are you referring to storing the connection, or storing the connection status?

Reply:
I've updated the signer interface to require a `ping()` method, and I've updated a bit of the interface and implementation code based on some recent corrections to my understanding of NIP-46.

Regarding checking connection status, I think each instantiation of a class that implements `INostrConnectSigner` will connect to one and only one signer.  A client could choose to juggle multiple signers, and if it does, it will have to do so by tracking multiple `INostrConnectSigner` instances.  I'm sticking to one connection per signer, as that is typical practice.